contents/blog/clickhouse-vs-bigquery.md

```diff
@@ -0,0 +1,168 @@
+
+BigQuery is exclusively available on GCP (Google Cloud Platform) and is tightly integrated with other GCP products. However, BigQuery can also connect to AWS and Azure storage through [BigQuery Omni](https://cloud.google.com/blog/products/data-analytics/introducing-bigquery-omni), though that defeats some of BigQuery’s all-in-one value-prop.  
+
+One of the reasons that BigQuery touts a feature-rich and friendly user-interface is to appeal to marketers with some basic SQL skills. While BigQuery isn’t as non-technical friendly as [Kibana](/blog/clickhouse-vs-elasticsearch) or [Snowflake](/blog/clickhouse-vs-snowflake), it does make using SQL easier by providing a GUI wrapper with the ability to save progress. Of course, for development purposes, BigQuery can also be accessed via GCP’s command line suite. 
+
+## Architecture
+
+### BigQuery’s Architecture
+
+BigQuery’s serverless architecture was novel at the time of its 2010 debut. Every BigQuery instance has variable storage units (distributed replicas) connected to variable compute units (calling slots); as an instance’s demands grow, so does its utilization of the storage and compute slots. Even today, this approach stands out — most databases parallelize compute, but not storage. 
+
+![BigQuery Architecture.png](../images/blog/clickhouse-vs-bigquery/bigquery-architecture.png)
+
+BigQuery extends many Google projects to make this happen. Specifically, BigQuery uses Colossus, Google’s file storage system; Borg, Google’s job scheduler; Dremel, a multi-tenant cluster that executes the queries; and Jupiter, Google’s pentabit network. BigQuery works because Google has a massive general-purpose infrastructure designed for isolated storage and compute.
+
+![Google Infra.png](../images/blog/clickhouse-vs-bigquery/google-infra.png)
+
+Amid being built on titanic infrastructure, BigQuery usually returns results in the order of _seconds_, not _milliseconds_, because it focuses on scalability, not per-query performance. 
+
+### ClickHouse’s (Traditional) Architecture
+
+ClickHouse utilizes a shared-nothing architecture where CPU, Storage, and Memory are stored in a single, beefy machine. While ClickHouse *does* support sharding databases across many physical instances managed under Apache Zookeeper, its fundamental design is monolithic. When you use ClickHouse (traditionally), everything is happening in the same place. 
+
+![Shared Nothing Architecture.png](../images/blog/clickhouse-vs-bigquery/shared-nothing-architecture.png)
+
+Accordingly, ClickHouse can scale in all three dimensions (storage, memory, and compute) by simply upgrading the machine. However, scaling is still tedious — new storage and compute cannot be added on a whim to the same instance; instead, a new instance needs to be provisioned and data needs to be migrated from one to another. 
```

Comment:
Not exactly true. You can always add storage to CH by adding disks to the instance https://altinity.com/blog/2019/11/27/amplifying-clickhouse-capacity-with-multi-volume-storage-part-1

definitely not as elegant as bigquery's solution though